src/hotspot/share/oops/methodData.cpp

```diff
@@ -59,9 +59,12 @@ bool DataLayout::needs_array_len(u1 tag) {
 // Perform generic initialization of the data.  More specific
 // initialization occurs in overrides of ProfileData::post_initialize.
 void DataLayout::initialize(u1 tag, u2 bci, int cell_count) {
-  _header._bits = (intptr_t)0;
-  _header._struct._tag = tag;
-  _header._struct._bci = bci;
+  DataLayout temp;
+  temp._header._bits = (intptr_t)0;
+  temp._header._struct._tag = tag;
+  temp._header._struct._bci = bci;
+  _header = temp._header;  // Write the cell atomtically
```

Comment:
Should we use `Atomic::store()` here?

Reply:
I don't think it's necessary.  It just needs to write the whole value once instead of performing 3 writes of differing sizes to the same cell.  The value being written is always that same as the value that was already there from the original initialization.  Maybe `atomically` is the wrong comment.  Maybe `Write the cell as an intptr_t unit`?